src/sources/mysql/mysql-schema.lisp

```diff
@@ -4,57 +4,29 @@
-;;;
-(defun create-my-views (views-alist)
-  "VIEWS-ALIST associates view names with their SQL definition, which might
-   be empty for already existing views. Create only the views for which we
-   have an SQL definition."
-  (unless (eq :all views-alist)
-    (let ((views (remove-if #'null views-alist :key #'cdr)))
-      (when views
-        (loop :for (name . def) :in views
-           :for sql := (destructuring-bind (schema . v-name) name
-                         (format nil
-                                 "CREATE VIEW ~@[~s.~]~a AS ~a"
-                                 schema v-name def))
-           :do
-           (log-message :info "MySQL: ~a" sql)
-           #+pgloader-image
-           (mysql-query sql)
-           #-pgloader-image
-           (restart-case
-               (mysql-query sql)
-             (use-existing-view ()
-               :report "Use the already existing view and continue"
-               nil)
-             (replace-view ()
-               :report "Replace the view with the one from pgloader's command"
-               (let* ((v-name   (cdr name))
-                      (drop-sql (format nil "DROP VIEW ~a;" v-name)))
-                 (log-message :info "MySQL: ~a" drop-sql)
-                 (mysql-query drop-sql)
-                 (mysql-query sql)))))))))
-
-(defun drop-my-views (views-alist)
-  "See `create-my-views' for VIEWS-ALIST description. This time we DROP the
-   views to clean out after our work."
-  (unless (eq :all views-alist)
-    (let ((views (remove-if #'null views-alist :key #'cdr)))
-      (when views
-        (let ((sql
-               (format nil "DROP VIEW ~{~a~^, ~};"
-                       (mapcar (lambda (qname)
-                                 (format nil "~@[~s.~]~a"
-                                         (car qname) (cdr qname)))
-                               (mapcar #'car views)))))
-          (log-message :info "MySQL: ~a" sql)
-          (mysql-query sql))))))
+(defclass copy-mysql (db-copy)
+  ((encoding :accessor encoding         ; allows forcing encoding
+             :initarg :encoding
+             :initform nil)
+   (range-list :accessor range-list
+               :initarg :range-list
+               :initform nil))
+  (:documentation "pgloader MySQL Data Source"))
+
+(defmethod initialize-instance :after ((source copy-mysql) &key)
+  "Add a default value for transforms in case it's not been provided."
+  (let ((transforms (and (slot-boundp source 'transforms)
+                         (slot-value  source 'transforms))))
+    (when (and (slot-boundp source 'fields) (slot-value source 'fields))
+      ;; cast typically happens in copy-database in the schema structure,
+      ;; and the result is then copied into the copy-mysql instance.
+      (unless (and (slot-boundp source 'columns) (slot-value source 'columns))
+        (setf (slot-value source 'columns)
+              (mapcar #'cast (slot-value source 'fields))))
+
+      (unless transforms
```

Comment:
The variable `TRANSFORMS` should only be declared here, since we are not using it in the above `WHEN UNLESS SETF` form.

Reply:
Code removed.